src/window.rs

```diff
@@ -205,12 +222,14 @@ impl Window {
     ) -> Result<Window> {
         let event_loop = EventsLoop::new();
 
+        let title = options.title.as_ref().map_or(DEFAULT_TITLE as &str, |t| &t);
+        let class = options.class.as_ref().map_or(DEFAULT_CLASS as &str, |c| &c);
```

Comment:
The `as &str` *might* not be needed.

Reply:
As you've already pointed out, the `as &str` should not be necessary.

Reply:
In fact, it's even possible to change it from `|t| &t` to `|t| t`.

Reply:
I actually hate that last option, just because it highlights the fact that this is all a kinda unfortunate hack.